core/src/main/java/software/amazon/awssdk/core/SdkResponse.java

```diff
@@ -26,6 +26,20 @@
  */
 @Immutable
 public abstract class SdkResponse {
+
+    private final SdkHttpData sdkHttpData;
+
+    protected SdkResponse(Builder builder) {
+        this.sdkHttpData = builder.sdkHttpData();
+    }
+
+    /**
+     * @return the optional data about the HTTP response.
```

Comment:
Why is it optional? Are there cases where we won't be able to provide it?

Reply:
Yeah, looks like not, I will remove it.

Reply:
Could be nice if we add other protocols in the future? We might want to figure out whether SdkResponse is going to be SdkHttpResponse in the future or if we'll want to mix in other protocols a different way.